### PR TITLE
[MM-18975] Migrate 'components/confirm_modal.jsx' to TypeScript

### PR DESCRIPTION
**Pull request by bolariin**
#### Summary
This PR migrates 'components/confirm_modal.jsx' and associated tests to TypeScript

#### Ticket Link
Resolves https://github.com/mattermost/mattermost-server/issues/12453

### CODE REVIEW

File: components/confirm_modal.tsx [hunk 1]
```diff
@@ -1,80 +1,83 @@
...
+
+    /*
+     * Text/jsx element to display with the checkbox
+     */
+    checkboxText?: React.ReactNode;
+
+    /*
+     * Function called when the confirm button or ENTER is pressed. Passes `true` if the checkbox is checked
+     */
+    onConfirm: Function;
```
Comment (devinbinnie) [inline components/confirm_modal.tsx]: Can we set the type for this one and the one below it to `(checked: boolean) => void`?
```suggestion
    onConfirm: (checked: boolean) => void;
```

File: components/confirm_modal.tsx [hunk 2]
```diff
@@ -1,80 +1,83 @@
...
+
+    /*
+     * Function called when the cancel button is pressed or the modal is hidden. Passes `true` if the checkbox is checked
+     */
+    onCancel: Function;
```
Comment (devinbinnie) [inline components/confirm_modal.tsx]:
```suggestion
    onCancel: (checked: boolean) => void;
```

**Comment by jgilliam17** [comment]
@devinbinnie can you please take a look and help create a test server? I am experiencing a high failure rate today 🙂

**Comment by jgilliam17** [comment]
@devinbinnie 
Can you please take a look at Main Menu>Manage Members modal? (as sysadmin)
It doesn't open for me, just loads blank page

**Comment by devinbinnie** [comment]
@jgilliam17 Seems to be working now, I think there was a bad merge at some point.

**Pull request merged**